### PR TITLE
DBZ-720 Extracting HistorizedRelationalSnapshotChangeEventSource

### PR DESCRIPTION
**Pull request by gunnarmorling**
Requires https://github.com/debezium/debezium/pull/569

@jpechane Here's the counterpart to the core PR.

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java [hunk 1]
```diff
@@ -6,229 +6,144 @@
...
-        }
+    @Override
+    protected Set<TableId> getAllTableIds(SnapshotContext ctx) throws Exception {
+        return jdbcConnection.readTableNames(ctx.catalogName, null, null, new String[] {"TABLE"} );
     }
 
-    private void determineCapturedTables(SnapshotContext ctx) throws SQLException {
-        Set<TableId> allTableIds = jdbcConnection.readTableNames(ctx.catalogName, null, null, new String[] {"TABLE"} );
+    @Override
+    protected boolean lockTables(ChangeEventSourceContext sourceContext, SnapshotContext snapshotContext) throws SQLException {
```
Comment (jpechane) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java]: Would it be possible to pull this method up and introduce an abstract method lockTable()?

**Comment by jpechane** [comment]
@gunnarmorling One comment add, otherwise it is good to go

**Comment by gunnarmorling** [comment]
I was wondering that, too. But then I thought some DBs might be able to
lock all tables at once (e.g. MySQL), so it seemed preferable to let
implementations iterate themselves. But we might re-consider based on your
other question around a "lockDatabase()" method. I think we should adjust
this step by step, as we add more implementations. Knowing the specific
needs, it'll be easier than trying to design it upfront.

**Comment by jpechane** [comment]
@gunnarmorling ok, I'll probably do both thinks when I use it for SQL server :-)

**Comment by jpechane** [comment]
@gunnarmorling Appled, thanks!

**Pull request merged**